pkg/bucket/client.go

```diff
@@ -42,167 +38,31 @@ func NewClient(b v1alpha1.CloudStorage, c client.Client) (Client, error) {
-		return err
-	}
-	deleteInput := &s3.DeleteBucketTaggingInput{Bucket: aws.String(a.bucket.Spec.Name)}
-	_, err = s3Client.DeleteBucketTagging(deleteInput)
-	if err != nil {
-		return err
-	}
-	input := CreateBucketTaggingInput(a.bucket.Spec.Name, a.bucket.Spec.Tags)
-
-	_, err = s3Client.PutBucketTagging(input)
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
-// CreateBucketTaggingInput creates an S3 PutBucketTaggingInput object,
-// which is used to associate a list of tags with a bucket.
-func CreateBucketTaggingInput(bucketname string, tags map[string]string) *s3.PutBucketTaggingInput {
-	putInput := &s3.PutBucketTaggingInput{
-		Bucket: aws.String(bucketname),
-		Tagging: &s3.Tagging{
-			TagSet: []*s3.Tag{},
-		},
-	}
-	for key, value := range tags {
-		newTag := s3.Tag{
-			Key:   aws.String(key),
-			Value: aws.String(value),
-		}
-		putInput.Tagging.TagSet = append(putInput.Tagging.TagSet, &newTag)
-	}
-	return putInput
-}
-
-func (a awsBucketClient) getS3Client() (s3iface.S3API, error) {
-	awsConfig := &aws.Config{Region: &a.bucket.Spec.Region}
-	cred, err := a.getCredentialFromSecret()
-	if err != nil {
-		return nil, err
-	}
-
-	opts := session.Options{
-		Config:            *awsConfig,
-		SharedConfigFiles: []string{cred},
-	}
-
-	if a.bucket.Spec.EnableSharedConfig != nil && *a.bucket.Spec.EnableSharedConfig {
-		opts.SharedConfigState = session.SharedConfigEnable
-	}
-
-	s, err := session.NewSessionWithOptions(opts)
-	if err != nil {
-		return nil, err
-	}
-	return s3.New(s), nil
-}
-
-func (a awsBucketClient) getCredentialFromSecret() (string, error) {
+func getCredentialFromSecret(a Client) (string, error) {
 	var filename string
 	var ok bool
-	namespaceName := types.NamespacedName{Namespace: a.bucket.Namespace, Name: a.bucket.Name}
+	namespaceName := types.NamespacedName{Namespace: a.getCloudStorage().Namespace, Name: a.getCloudStorage().Name}
 	if filename, ok = fileBucketCache[namespaceName]; !ok {
 		// Look for file in tmp based on name.
 		// TODO: handle force credential refesh
 		secret := &corev1.Secret{}
-		err := a.client.Get(context.TODO(),
+		err := a.getClient().Get(context.TODO(),
 			types.NamespacedName{
-				Name:      a.bucket.Spec.CreationSecret.Name,
-				Namespace: a.bucket.Namespace,
+				Name:      a.getCloudStorage().Spec.CreationSecret.Name,
+				Namespace: a.getCloudStorage().Namespace,
 			},
 			secret)
 		if err != nil {
 			return "", err
 		}
 
-		cred := secret.Data[a.bucket.Spec.CreationSecret.Key]
+		cred := secret.Data[a.getCloudStorage().Spec.CreationSecret.Key]
 		//create a tmp file based on the bucket name, if it does not exist
-		dir, err := os.MkdirTemp("", fmt.Sprintf("secret-%v-%v", a.bucket.Namespace, a.bucket.Name))
+		dir, err := os.MkdirTemp("", fmt.Sprintf("secret-%v-%v", a.getCloudStorage().Namespace, a.getCloudStorage().Name))
 		if err != nil {
 			return "", err
 		}
-		f, err := os.CreateTemp(dir, "aws-secret")
+		f, err := os.CreateTemp(dir, "cloudstoragesecret")
```

Comment:
Get better names. (if this function is to be reused).

For multiple cloud providers might need to use a different name to not overwrite each other.

Reply:
The names could be passed in by each cloud provider's methods as another parameter.

Reply:
Since it's already in its own unique directory based on `cloudStorage Namespace` and `cloudStorage.Name` we don't need to parameterize this string. Can keep as is for now.